### ISSUE TITLE
description on list element aren't read by screen reader

### ISSUE THREAD

**Issue by alex19EP** [microsoft/vscode#117979]
Issue Type: <b>Bug</b>

1. use a screen reader
2. open extensions view
3. navigate the list

### what happens?

extension name, version and author is read by nvda and orca. extension  description isn't.

### what should happen?

extension name, version, description and author should be read by screen reader.

VS Code version: Code - Insiders 1.54.0-insider (e7989863202eac0ec4e66ca82733d968bca7527a, 2021-03-01T21:11:33.783Z)
OS version: Windows_NT x64 10.0.19043

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-7700K CPU @ 4.20GHz (8 x 4200)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>protected_video_decode: enabled<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.86GB (6.70GB free)|
|Process Argv|--crash-reporter-id cfcd7f4f-ba55-4a11-af03-c5d167f85294|
|Screen Reader|yes|
|VM|25%|
</details><details><summary>Extensions (45)</summary>

Extension|Author (truncated)|Version
---|---|---
meson|asa|1.3.0
github-markdown-preview|bie|0.0.2
markdown-checkbox|bie|0.1.3
markdown-emoji|bie|0.1.0
markdown-preview-github-styles|bie|0.1.6
markdown-yaml-preamble|bie|0.0.4
vscode-astyle|chi|0.9.0
systemd-unit-file|coo|1.0.6
vscode-github-actions|csc|0.16.0
doxdocgen|csc|1.1.0
vscode-markdownlint|Dav|0.39.0
githistory|don|0.6.14
gitlens|eam|11.2.1
EditorConfig|Edi|0.16.4
vscode-pull-request-github-insiders|Git|2021.3.1984
vscode-test-explorer|hbe|2.19.5
better-cpp-syntax|jef|1.15.5
restructuredtext|lex|145.0.0
kconfig|luv|1.0.3
vscode-docker|ms-|1.10.0
vscode-language-pack-ru|MS-|1.54.1
python|ms-|2021.3.593898838-dev
vscode-pylance|ms-|2021.2.4
jupyter|ms-|2021.3.600686576
remote-containers|ms-|0.163.0
remote-ssh|ms-|0.64.0
remote-ssh-edit|ms-|0.64.0
remote-wsl|ms-|0.54.0
cmake-tools|ms-|1.6.0
cpptools|ms-|1.2.2
cpptools-extension-pack|ms-|1.0.0
powershell-preview|ms-|2021.2.1
vsliveshare|ms-|1.0.3784
vsliveshare-audio|ms-|0.1.91
vsliveshare-pack|ms-|0.4.0
vsonline|ms-|1.0.3076
awk-vscode|pet|0.0.1
vscode-yaml|red|0.15.0
platform-settings|run|0.0.2
vscode-table-formatter|shu|1.2.1
rewrap|stk|1.14.0
code-spell-checker|str|1.10.2
code-spell-checker-russian|str|0.2.9
cmake|twx|0.0.17
grammarly|znc|0.14.0

</details><details>
<summary>A/B Experiments</summary>

```
vsliv695:30137379
vsins829:30139715
vsliv368:30146709
vsreu685:30147344
python383cf:30185419
pythonvspyt602:30263608
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492:30256197
pythonvsdeb440:30224570
pythonvsded773:30223139
pythonvspyt875:30259475
pythonvsnew554:30265444
pythontb:30258533
vscoresta800:30265947

```

</details>

**Comment by isidorn** [comment]
I just checked this and yeah by design we decided to make the aria label
NAME, VERSION, PUBLISHER, press enter for extension details

I suggest that we rename it to:
NAME, VERSION, PUBLISHER, DESCRIPTION, press enter for more details

What do you think?

fyi @sandy081 in case you have thoughts here, otherwise I plan to do the change

**Comment by jvesouza** [comment]
Sounds good to me.

**Comment by alex19EP** [comment]
> NAME, VERSION, PUBLISHER, DESCRIPTION, press enter for more details

it seems to me that press enter for more details is superfluous information.
also I'm not sure if the order should be like this.

in fact there is also another problem similar to this one.

1.clone the repository with submodules. eg https://github.com/RHVoice/RHVoice/
2. open git view.

### what happens?

screen reader reads only git on each repo.

### what should happen?

screen reader should read name of repo.

**Comment by isidorn** [comment]
@alex19EP those are great catches, thanks a lot!
I have pushed a fix both for the extensions view and the scm view, try it out and let me know what you think

fyi @eamodio for my minor scm changes

**Comment by sandy081** [comment]
@isidorn Your changes makes sense to me in extensions land. Thanks for changes.

**Comment by jvesouza** [comment]
Working as expected. The orca now reads the description of the extension along with the other information.
Great job.

**Comment by isidorn** [comment]
@jvesouza great, thanks for letting us know. Adding verified label.

**Comment by eamodio** [comment]
@isidorn Looks good -- thanks!

**Comment by alex19EP** [comment]
FYI @isidorn if I use Russian as code interface language ext description isn't read by screen reader.
instead it says press enter for more details. I think language pack needs to somehow untranslate this string.

**Comment by isidorn** [comment]
@alex19EP thanks for letting us know. It usually takes some time for the language pack to update to the new string.
However I changed the id of the nls string, so it should hopefully update in insiders tomorrow. If not, then the language team needs to pick this up which will take some time
